src/memex/markdown.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+
+import bleach
+import mistune
+
+MARKDOWN_TAGS = [
+    'a', 'blockquote', 'code', 'em', 'hr', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6',
+    'img', 'li', 'ol', 'p', 'pre', 'strong', 'ul',
+]
+ALLOWED_TAGS = set(bleach.ALLOWED_TAGS + MARKDOWN_TAGS)
+
+MARKDOWN_ATTRIBUTES = {
+    'a': ['href', 'title'],
+    'img': ['alt', 'src', 'title'],
+}
+ALLOWED_ATTRIBUTES = dict(bleach.ALLOWED_ATTRIBUTES.items() + MARKDOWN_ATTRIBUTES.items())
+
+markdown = None
+
+
+class MathMarkdown(mistune.Markdown):
+    def output_block_math(self):
+        return self.renderer.block_math(self.token['text'])
+
+
+class MathInlineLexer(mistune.InlineLexer):
+    def __init__(self, *args, **kwargs):
+        super(MathInlineLexer, self).__init__(*args, **kwargs)
+        self.rules.inline_math = re.compile(r'\\\((.*?)\\\)', re.DOTALL)
+        self.default_rules.insert(0, 'inline_math')
+
+    def output_inline_math(self, m):
+        return self.renderer.inline_math(m.group(1))
+
+
+class MathBlockLexer(mistune.BlockLexer):
+    def __init__(self, *args, **kwargs):
+        super(MathBlockLexer, self).__init__(*args, **kwargs)
+        self.rules.block_math = re.compile(r'^\$\$(.*?)\$\$', re.DOTALL)
+        self.default_rules.insert(0, 'block_math')
+
+    def parse_block_math(self, m):
+        self.tokens.append({
+            'type': 'block_math',
+            'text': m.group(1)
+        })
+
+
+class MathRenderer(mistune.Renderer):
+    def __init__(self, **kwargs):
+        self.options = kwargs
```

Comment:
Should this call the base class constructor of `mistune.Renderer` here, passing `kwargs`?